lib/galaxy/security/passwords.py

```diff
@@ -42,16 +44,28 @@ def hash_password_PBKDF2(password):
     salt = b64encode(urandom(SALT_LENGTH))
     # Apply the pbkdf2 encoding
     hashed = pbkdf2_bin(smart_str(password), salt, COST_FACTOR, KEY_LENGTH, getattr(hashlib, HASH_FUNCTION))
+    hashed_b64 = b64encode(hashed)
+    if six.PY3:
+        salt = salt.decode('utf-8')
+        hashed_b64 = hashed_b64.decode('utf-8')
     # Format
-    return 'PBKDF2${0}${1}${2}${3}'.format(HASH_FUNCTION, COST_FACTOR, salt, b64encode(hashed))
+    return 'PBKDF2${0}${1}${2}${3}'.format(HASH_FUNCTION, COST_FACTOR, salt, hashed_b64)
 
 
 def check_password_PBKDF2(guess, hashed):
     # Split the database representation to extract cost_factor and salt
     name, hash_function, cost_factor, salt, encoded_original = hashed.split('$', 5)
+    if six.PY3:
```

Comment:
It's already merged, and I'm sure this works, but it seemed like it was simpler to just use smart_str (as in the pbkdf2-specific PR) on the parts without this extra conditional logic.  Did that approach not work for the salt?

Reply:
Let me try that and let you know.

Reply:
Ok.  Want me to do it, or do you have a commit ready to go?  I'm happy to merge the tweaks making this more concise and handling it just with smart_str for 18.05 if you want to PR it.

Reply:
I don't have anything ready to go, if you want to make that change that would be great. Thanks!

Reply:
@jmchilton Looks like the release already branched, so I'm just going to target dev with a refactor (probably tomorrow, unless you want to do it before), no worries.